src/pybind/mgr/dashboard/controllers/user.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+from . import ApiController, AuthRequired, RESTController
+from .. import mgr
+from .auth import Auth
+
+
+@ApiController('user')
+@AuthRequired()
+class User(RESTController):
+    # pylint: disable=unused-argument
+    def get(self, username):
+        config_username = mgr.get_config('username', None)
+
+        return {
+            'username': config_username
+        }
+
+    @RESTController.args_from_json
+    def set(self, username, password=None):
+        mgr.set_config('username', username)
+
+        if password:
+            hashed_passwd = Auth.password_hash(password)
```

Comment:
If we create a user management service (as stated in the previous comment) we can put this logic hidden inside the service and just expose a well defined public interface to update the username and password without revealing the implementation details.